src/main/xsl/preprocess/topicpullImpl.xsl

```diff
@@ -169,16 +261,20 @@ mode="topicpull:figure-linktext" and mode="topicpull:table-linktext"
   <!-- If an attribute is specified locally, set it. Otherwise, try to inherit from ancestors. -->
   <xsl:template match="*" mode="topicpull:inherit-from-self-then-ancestor">
     <xsl:param name="attrib"/>
-    <xsl:variable name="attrib-here">
-      <xsl:if test="@*[local-name()=$attrib]"><xsl:value-of select="@*[local-name()=$attrib]"/></xsl:if>
+    <xsl:variable name="attrib-here" as="xs:string?">
+      <xsl:if test="@*[local-name()=$attrib]">
```

Comment:
The `xsl:if` is redundant, the nested `xsl:value-of` already contains the condition.